src/net_processing.cpp

```diff
@@ -3856,7 +3856,7 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
         {
         LOCK(cs_main);
 
-        if (!m_chainman.m_blockman.LookupBlockIndex(cmpctblock.header.hashPrevBlock)) {
+        if (!LookupBlockIndexForPeer(pfrom, cmpctblock.header.hashPrevBlock)) {
```

Comment:
I might be not getting something... In previous commit, we `MaybeSendGetHeaders` if this condition is true. Here, we `MaybeSendGetHeaders` if the same condition is false. Seems strange, but maybe I'm missing the reasoning.

Reply:
Ah, I guess this branch assumes that this new header extends our best chain, and would try to build on it.

So the probe would be to send two different compact blocks: one on top of the stale block, another one not on top of it. After this change, it would make a difference only if the stale block is registered by the peer's network id. Alright, makes sense. Feel free to resolve.

Reply:
Could benefit from a comment :)

Reply:
Added some comments here